Content.Server/_DV/CosmicCult/Abilities/CosmicFragmentationSystem.cs

```diff
@@ -0,0 +1,148 @@
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly PolymorphSystem _polymorph = default!;
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<AILawUpdatedEvent>(OnLawInserted);
+
+        SubscribeLocalEvent<CosmicCultComponent, EventCosmicFragmentation>(OnCosmicFragmentation);
+        SubscribeLocalEvent<CosmicCultComponent, EventCosmicChantryDoAfter>(OnCosmicChantryDoAfter);
+        SubscribeLocalEvent<CosmicCultComponent, EventCosmicFragmentationDoAfter>(OnCosmicFragmentationDoAfter);
+    }
+
+    private void UnEmpower(Entity<CosmicCultComponent> ent)
+    {
+        ent.Comp.CosmicEmpowered = false; // empowerment spent! Now we set all the values back to their default.
+        ent.Comp.CosmicSiphonQuantity = 1;
+        ent.Comp.CosmicGlareRange = 8;
+        ent.Comp.CosmicGlareDuration = TimeSpan.FromSeconds(5);
+        ent.Comp.CosmicGlareStun = TimeSpan.FromSeconds(0);
+        ent.Comp.CosmicImpositionDuration = TimeSpan.FromSeconds(5.8);
+        ent.Comp.CosmicBlankDuration = TimeSpan.FromSeconds(22);
+        ent.Comp.CosmicBlankDelay = TimeSpan.FromSeconds(0.6);
+    }
+
+    private void OnCosmicFragmentation(Entity<CosmicCultComponent> ent, ref EventCosmicFragmentation args)
+    {
+        if (args.Handled || HasComp<ActiveNPCComponent>(args.Target) || TryComp<MobStateComponent>(args.Target, out var state) && state.CurrentState != MobState.Alive)
+        {
+            _popup.PopupEntity(Loc.GetString("cosmicability-generic-fail"), ent, ent);
+            return;
+        }
+        if (HasComp<EmagSiliconLawComponent>(args.Target))
+        {
+            var doargs = new DoAfterArgs(EntityManager, ent, ent.Comp.CosmicSiphonDelay, new EventCosmicChantryDoAfter(), ent, args.Target)
+            {
+                DistanceThreshold = 2f,
+                Hidden = false,
+                BreakOnHandChange = true,
+                BreakOnDamage = true,
+                BreakOnMove = true,
+                BreakOnDropItem = true,
+            };
+            args.Handled = true;
+            _doAfter.TryStartDoAfter(doargs);
+        }
+        else if (HasComp<SiliconLawUpdaterComponent>(args.Target))
+        {
+            var doargs = new DoAfterArgs(EntityManager, ent, ent.Comp.CosmicSiphonDelay, new EventCosmicFragmentationDoAfter(), ent, args.Target)
+            {
+                DistanceThreshold = 2f,
+                Hidden = false,
+                BreakOnHandChange = true,
+                BreakOnDamage = true,
+                BreakOnMove = true,
+                BreakOnDropItem = true,
+            };
+            args.Handled = true;
+            _doAfter.TryStartDoAfter(doargs);
+        }
```

Comment:
1. DRY
1. You have a whitelist of components in YML but hardcode the actions here, that's a bad mix